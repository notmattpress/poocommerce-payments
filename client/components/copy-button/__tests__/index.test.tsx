/** @format **/

/**
 * External dependencies
 */
import React from 'react';
import { act, fireEvent, render, screen } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { CopyButton } from '..';

describe( 'CopyButton', () => {
	it( 'renders the button correctly', () => {
		const { container: copyButtonContainer } = render(
			<CopyButton
				textToCopy="test_bank_reference_id"
				label="Copy bank reference ID to clipboard"
			/>
		);

		expect( copyButtonContainer ).toMatchSnapshot();
	} );

	describe( 'when the button is clicked', () => {
		it( 'copies the text to the clipboard and shows copied state', async () => {
			render(
				<CopyButton
					textToCopy="test_bank_reference_id"
					label="Copy bank reference ID to clipboard"
				/>
			);

			const button = screen.queryByRole( 'button', {
				name: /Copy bank reference ID to clipboard/i,
			} );

			if ( ! button ) {
				throw new Error( 'Button not found' );
			}

			//Mock the clipboard API
			Object.assign( navigator, {
				clipboard: {
					writeText: jest.fn().mockResolvedValueOnce( undefined ),
				},
			} );

			await act( async () => {
				fireEvent.click( button );
			} );

			expect( navigator.clipboard.writeText ).toHaveBeenCalledWith(
				'test_bank_reference_id'
			);
			expect( button ).toHaveClass( 'state--copied' );

			act( () => {
				fireEvent.animationEnd( button );
			} );

			expect( button ).not.toHaveClass( 'state--copied' );
		} );
	} );
} );
