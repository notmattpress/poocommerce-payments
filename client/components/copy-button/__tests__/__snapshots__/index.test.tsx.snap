// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyButton renders the button correctly 1`] = `
<div>
  <button
    aria-label="Copy bank reference ID to clipboard"
    class="woopayments-copy-button"
    title="Copy to clipboard"
    type="button"
  >
    <i />
  </button>
</div>
`;
